Rate Limiting docs (#1877)
linkerd/website: linkerd.io/content/2-edge/reference/rate-limiting.md

---
title: Rate Limiting
description: Reference guide to Linkerd's HTTPLocalRateLimitPolicy resource
---

Linkerd's rate limiting functionality is configured via
`HTTPLocalRateLimitPolicy` resources, which should point to a
[Server](../../reference/authorization-policy/#server) reference. Note that a
`Server` can only be referred by a single `HTTPLocalRateLimitPolicy`.

{{< note >}}
`Server`'s default `accessPolicy` config is `deny`. This means that if you don't
have [AuthorizationPolicies](../../reference/authorization-policy/) pointing to a
Server, it will deny traffic by default. If you want to set up rate limit
policies for a Server without being forced to also declare authorization
policies, make sure to set `accessPolicy` to a permissive value like
`all-unauthenticated`.
{{< /note >}}

## HTTPLocalRateLimitPolicy Spec

{{< keyval >}}
| field| value |
|------|-------|
| `targetRef`| A reference to the [Server](../../reference/authorization-policy/#server) this policy applies to. |

> | `total.requestsPerSecond`| Overall rate limit, which all traffic coming to the `targetRef` should abide. If unset no overall limit is applied. |

Reviewer: ```suggestion
| `total.requestsPerSecond`| Overall rate limit for all traffic sent to the `targetRef`. If unset no overall limit is applied. |
```

> | `total.identity`| Fairness for individual identities; each separate client, grouped by identity, will have this rate limit. The `requestsPerSecond` value should be less than or equal to `total.requestsPerSecond` (if set). |

Reviewer: ```suggestion
| `identity.requestsPerSecond`| Fairness for individual identities; each separate client, grouped by identity, will have this rate limit. If `total.requestsPerSecond` is also set, `identity.requestsPerSecond` cannot be greater than `total.requestsPerSecond`. |
```

Author: Good find! :sweat_smile:

> | `overrides`| An array of [overrides](#overrides) for traffic from specific clients. The `requestsPerSecond` values should be less than or equal to `total.requestsPerSecond` (if set). |

Reviewer: ```suggestion
| `overrides`| An array of [overrides](#overrides) for traffic from specific client). |
```

{{< /keyval >}}

### Overrides

{{< keyval >}}
| field| value |
|------|-------|

> | `requestsPerSecond`| The number of requests per second allowed from clients matching `clientRefs`. |

Reviewer: ```suggestion
| `requestsPerSecond`| The number of requests per second allowed from clients matching `clientRefs`.  If `total.requestsPerSecond` is also set, the `requestsPerSecond` for each `overrides` entry cannot be greater than `total.requestsPerSecond`. |
```
